ios Project fails with `The hashAssetFiles Metro plugin is not configured`

## 🐛 Bug Report

On a fresh, bare typescript expo project, I get the error: `The hashAssetFiles Metro plugin is not configured. You need to add a metro.config.js to your project that configures Metro to use this plugin`

### Environment

<!-- In your terminal run `expo diagnostics` and paste the output here. -->
  Expo CLI 3.20.0 environment info:
    System:
      OS: macOS Mojave 10.14.6
      Shell: 3.2.57 - /bin/bash
    Binaries:
      Node: 10.19.0 - /usr/local/bin/node
      Yarn: 1.22.0 - ~/.yarn/bin/yarn
      npm: 6.13.4 - /usr/local/bin/npm
      Watchman: 4.9.0 - /usr/local/bin/watchman
    IDEs:
      Android Studio: 3.5 AI-191.8026.42.35.6010548
      Xcode: 11.3.1/11C505 - /usr/bin/xcodebuild
    npmPackages:
      expo: ~37.0.3 => 37.0.10 
      react: ~16.9.0 => 16.9.0 
      react-dom: ~16.9.0 => 16.9.0 
      react-native: ~0.61.5 => 0.61.5 
      react-native-web: ~0.11.7 => 0.11.7 
    npmGlobalPackages:
      expo-cli: 3.20.0

<!-- Please also let us know about your app's target (iOS, Android, Client, Standalone, ExpoKit) -->

### Steps to Reproduce

1. create the project `expo init` and choose unmanaged bare typescript template
2. run pod install from the ios folder 
3. run `yarn ios`

### Expected Behavior

yarn ios should build without incident

### Actual Behavior

Build fails with this message: `[redacted]`

### Reproducible Demo

~https://github.com/joshglenn/expo-issue-may-13~ this does not reproduce the reported issue

@byCedric @brentvatne I can confirm that a fresh install as illustrated in https://github.com/expo/expo/issues/8294#issuecomment-629037499 is affected by this problem. 
I run into it while ejecting a fairly new expo app. Updating to latest expo, expo-cli and expo-updates did not fix the issue.

It looks like it is related to https://github.com/expo/expo/pull/8324. That PR rely on `@react-native-community/cli` which is not listed as a dependency on freshly ejected project. 

On clean project `npm run ios` results with:
```
Error: Cannot find module '@react-native-community/cli/build/commands/bundle/filterPlatformAssetScales'
Require stack:
- /Users/chrmod/Projects/game/node_modules/expo-updates/scripts/createManifest.js
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:980:15)
    at Function.Module._load (internal/modules/cjs/loader.js:862:27)
    at Module.require (internal/modules/cjs/loader.js:1042:19)
    at require (internal/modules/cjs/helpers.js:77:18)
    at Object.<anonymous> (/Users/chrmod/Projects/game/node_modules/expo-updates/scripts/createManifest.js:1:35)
    at Module._compile (internal/modules/cjs/loader.js:1156:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1176:10)
    at Module.load (internal/modules/cjs/loader.js:1000:32)
    at Function.Module._load (internal/modules/cjs/loader.js:899:14)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:74:12) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/Users/chrmod/Projects/game/node_modules/expo-updates/scripts/createManifest.js'
  ]
}
```

Installing `@react-native-community/cli` complains about:
```
The hashAssetFiles Metro plugin is not configured. You need to add a metro.config.js to your project that configures Metro to use this plugin
```

All problems go away after running `pod install`

--- a/packages/expo-updates/CHANGELOG.md
+++ b/packages/expo-updates/CHANGELOG.md
@@ -8,6 +8,8 @@
 
 ### 🐛 Bug fixes
 
+- Temporarily vendor `filterPlatformAssetScales` method from `@react-native-community/cli` in order to fix builds when `npm` was used to install dependencies (rather than `yarn`).
+
 ## 0.2.2
 
 ### 🐛 Bug fixes

--- a/packages/expo-updates/scripts/createManifest.js
+++ b/packages/expo-updates/scripts/createManifest.js
@@ -1,9 +1,9 @@
-const filterPlatformAssetScales = require('@react-native-community/cli/build/commands/bundle/filterPlatformAssetScales')
-  .default;
 const fs = require('fs');
 const http = require('http');
 const path = require('path');
 const uuid = require('uuid/v4');
+
+const filterPlatformAssetScales = require('./filterPlatformAssetScales');
 
 const platform = process.argv[2];
 const packagerUrl = process.argv[3];

--- a/packages/expo-updates/scripts/filterPlatformAssetScales.js
+++ b/packages/expo-updates/scripts/filterPlatformAssetScales.js
@@ -1,0 +1,33 @@
+// copied from https://github.com/react-native-community/cli/blob/48136adfb814d335e957e22129d049c4a05c8759/packages/cli/src/commands/bundle/filterPlatformAssetScales.ts
+
+const ALLOWED_SCALES = {
+  ios: [1, 2, 3],
+};
+
+function filterPlatformAssetScales(platform, scales) {
+  const whitelist = ALLOWED_SCALES[platform];
+  if (!whitelist) {
+    return scales;
+  }
+  const result = scales.filter(scale => whitelist.indexOf(scale) > -1);
+  if (result.length === 0 && scales.length > 0) {
+    // No matching scale found, but there are some available. Ideally we don't
+    // want to be in this situation and should throw, but for now as a fallback
+    // let's just use the closest larger image
+    const maxScale = whitelist[whitelist.length - 1];
+    for (const scale of scales) {
+      if (scale > maxScale) {
+        result.push(scale);
+        break;
+      }
+    }
+
+    // There is no larger scales available, use the largest we have
+    if (result.length === 0) {
+      result.push(scales[scales.length - 1]);
+    }
+  }
+  return result;
+}
+
+module.exports = filterPlatformAssetScales;

